src/node_perf.cc

```diff
@@ -554,6 +555,78 @@ bool ELDHistogram::Disable() {
   return true;
 }
 
+// Event Loop Timing Histogram
+namespace {
+static void RWLHistogramMin(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  double value = static_cast<double>(env->req_wrap_latency()->Min());
+  args.GetReturnValue().Set(value);
+}
+
+static void RWLHistogramMax(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  double value = static_cast<double>(env->req_wrap_latency()->Max());
+  args.GetReturnValue().Set(value);
+}
+
+static void RWLHistogramMean(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  args.GetReturnValue().Set(env->req_wrap_latency()->Mean());
+}
+
+static void RWLHistogramExceeds(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  double value = static_cast<double>(env->req_wrap_latency()->Exceeds());
+  args.GetReturnValue().Set(value);
+}
+
+static void RWLHistogramStddev(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  args.GetReturnValue().Set(env->req_wrap_latency()->Stddev());
+}
+
+static void RWLHistogramPercentile(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsNumber());
+  double percentile = args[0].As<Number>()->Value();
+  args.GetReturnValue().Set(env->req_wrap_latency()->Percentile(percentile));
+}
+
+static void RWLHistogramPercentiles(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsMap());
+  Local<Map> map = args[0].As<Map>();
+  env->req_wrap_latency()->Percentiles([&](double key, double value) {
+    map->Set(env->context(),
+             Number::New(env->isolate(), key),
+             Number::New(env->isolate(), value)).IsEmpty();
+  });
+}
+
+static void RWLHistogramEnable(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  bool enabled = env->req_wrap_latency_enabled();
+  if (!enabled) {
+    env->req_wrap_latency()->Reset();
+    env->set_req_wrap_latency_enabled(true);
+  }
+  args.GetReturnValue().Set(!enabled);
+}
+
+static void RWLHistogramDisable(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  bool enabled = env->req_wrap_latency_enabled();
+  if (enabled)
+    env->set_req_wrap_latency_enabled(false);
+  args.GetReturnValue().Set(enabled);
+}
+
+static void RWLHistogramReset(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  env->req_wrap_latency()->Reset();
+}
+}  // namespace
```

Comment:
Both this PR and https://github.com/nodejs/node/pull/26519 introduce a lot of these functions … can we avoid that and share some code?

Reply:
Yep, that's the long term intent... note that #26519 is still draft. I'm going to be reducing the duplication here and further reduce in that one. Before doing so, just wanted to make sure there were no objections to going this direction.